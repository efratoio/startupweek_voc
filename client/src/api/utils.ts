declare var baseUrl: string;

export function url(href: string) {
    return `${location.protocol}//${baseUrl}${href}`;
}

export interface ResponseWrapper<T> {
    readonly message: string;
    readonly okay: boolean;
    readonly data: T;
}

export async function parseApi<T>(unresolvedResponse: Promise<Response>): Promise<T> {
    let response: Response;
    try {
        response = await unresolvedResponse;
    } catch(err) {
        // TODO: Handle connection error.
        console.error(err);
        throw new Error("Unable to connect to the backend.");
    }
    if (response.status < 200 || response.status >= 300) {
        // TODO: Handle bad status code.
        throw new Error(`Bad status code ${response.status}.`);
    }
    let json: T;
    try {
        json = await response.json();
    } catch(err) {
        // TODO: Handle parse error.
        console.error(err);
        throw new Error("Unable to parse response.");
    }
    return json;
}
