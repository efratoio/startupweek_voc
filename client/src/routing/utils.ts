import * as patterns from "./patterns";

export type Route =
    { route: "login" } |
    { route: "dashboard" } |
    { route: "signup" };

export function parseRoute(url: string): Route {
    if (patterns.patternLogin.match(url)) {
        return { route: "login" };
    }
    if (patterns.patternDashboard.match(url)) {
        return { route: "dashboard" };
    }
    if (patterns.patternSignup.match(url)) {
        return { route: "signup" };
    }
}
