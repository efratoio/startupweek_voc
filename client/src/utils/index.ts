export * from "./environment";
export * from "./require-login";
export * from "./validation";
