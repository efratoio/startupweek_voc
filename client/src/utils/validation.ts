// Taken from http://emailregex.com/
const emailValidationRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/; // tslint:disable-line

export function validateEMail(email: string): boolean {
    return Boolean(email.match(emailValidationRegex));
}

export function validatePassword(password: string): boolean {
    return password.length >= 8;
}
