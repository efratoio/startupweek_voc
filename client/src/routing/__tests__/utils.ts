import { parseRoute } from "../utils";

["/login", "/signup", "/dashboard", "/unknown"].forEach(route => {
    test(`The route parsing utility detects "${route}" correctly`, () => {
        expect(parseRoute(route)).toMatchSnapshot();
    });
});
