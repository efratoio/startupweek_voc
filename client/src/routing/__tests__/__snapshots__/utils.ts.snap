// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The route parsing utility detects "/dashboard" correctly 1`] = `
Object {
  "route": "dashboard",
}
`;

exports[`The route parsing utility detects "/login" correctly 1`] = `
Object {
  "route": "login",
}
`;

exports[`The route parsing utility detects "/signup" correctly 1`] = `
Object {
  "route": "signup",
}
`;

exports[`The route parsing utility detects "/unknown" correctly 1`] = `undefined`;
